lib/rate_limited_request_map.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+function RateLimitedRequestMap() {
+  this._requests = new Map();
+}
+
+RateLimitedRequestMap.prototype.keepWorse = function add(rlRequest) {
```

Comment:
just read the code and I get what this is doing. I like it. we should document this function at least as it might be confusing